src/Microsoft.PowerShell.Commands.Utility/commands/utility/CsvCommands.cs

```diff
@@ -270,6 +270,14 @@ protected override void BeginProcessing()
                 this.ThrowTerminatingError(errorRecord);
             }
 
+            // Validate that Append and NoHeader are not specified together.
+            if (Append.IsPresent && NoHeader.IsPresent)
+            {
+                InvalidOperationException exception = new(CsvCommandStrings.CannotSpecifyAppendAndNoHeader);
+                ErrorRecord errorRecord = new(exception, "CannotSpecifyAppendAndNoHeader", ErrorCategory.InvalidArgument, null);
```

Comment:
Maybe (in test too):
```suggestion
                ErrorRecord errorRecord = new(exception, "CannotSpecifyBothAppendAndNoHeader", ErrorCategory.InvalidArgument, null);
```